bin/node/cli/src/service.rs

```diff
@@ -535,13 +536,16 @@ mod tests {
 
 				digest.push(<DigestItem as CompatibleDigestItem>::babe_pre_digest(babe_pre_digest));
 
-				let mut proposer = proposer_factory.init(&parent_header).unwrap();
-				let new_block = futures::executor::block_on(proposer.propose(
-					inherent_data,
-					digest,
-					std::time::Duration::from_secs(1),
-					RecordProof::Yes,
-				)).expect("Error making test block").block;
+				let proposing = proposer_factory.init(&parent_header)
+					.and_then(|mut proposer| proposer.propose(
+						inherent_data,
+						digest,
+						std::time::Duration::from_secs(1),
+						RecordProof::Yes,
+					));
+
+				let new_block = futures::executor::block_on(proposing)
```

Comment:
May I suggest something like this instead?

```suggestion
				let new_block = futures::executor::block_on(async move {
				    let proposer = proposer_factory.init(&parent_header).await;
				    proposer.propose(
						inherent_data,
						digest,
						std::time::Duration::from_secs(1),
						RecordProof::Yes,
					).await
				});
```

Reply:
still need to get the `async/await` into my bloodstream :)